mito-ai/src/Extensions/InlineCompleter/index.ts

```diff
@@ -0,0 +1,147 @@
+import type {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { Notification } from '@jupyterlab/apputils';
+import { IEditorLanguageRegistry } from '@jupyterlab/codemirror';
+import { ICompletionProviderManager } from '@jupyterlab/completer';
+import { ConfigSection } from '@jupyterlab/services';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+import { MitoAIInlineCompleter } from './provider';
+
+/**
+ * Interface for the Mito AI configuration settings.
+ */
+interface IMitoAIConfig {
+  state: {
+    /**
+     * Whether the settings have been checked or not.
+     */
+    settingsChecked: boolean;
+  };
+}
+
+const JUPYTERLAB_INLINE_COMPLETER_ID =
+  '@jupyterlab/completer-extension:inline-completer';
+export const completionPlugin: JupyterFrontEndPlugin<void> = {
+  id: 'mito-ai:inline-completion',
+  autoStart: true,
+  requires: [
+    ICompletionProviderManager,
+    IEditorLanguageRegistry,
+    ISettingRegistry
+  ],
+  activate: (
+    app: JupyterFrontEnd,
+    completionManager: ICompletionProviderManager,
+    languageRegistry: IEditorLanguageRegistry,
+    settingRegistry: ISettingRegistry
+  ) => {
+    if (typeof completionManager.registerInlineProvider === 'undefined') {
+      // Gracefully short-circuit on JupyterLab 4.0 and Notebook 7.0
+      console.warn(
+        'Inline completions are only supported in JupyterLab 4.1+ and Jupyter Notebook 7.1+'
+      );
+      return;
+    }
+
+    // Only alphanumeric characters are authorized for configuration section
+    const CONFIG_SECTION = 'mitoaiconfig';
+
+    // Use Jupyter Server configuration to check if this is the first time the extension is installed.
+    ConfigSection.create({
+      name: CONFIG_SECTION,
+      serverSettings: app.serviceManager.serverSettings
+    })
+      .then(async config => {
```

Comment:
This one is an optimization. An activation method can be asynchronous. But as this extension is auto-started at the application launch, it will be part of the extensions _blocking_ the start of the application. Therefore this is a good practice to carry network request as background task as done here. If from a style point of view, you prefer using `try {} catch () {}` syntax for such case. That could be rewritten using the following pattern:

```ts
const f = async () => {
    try {
        await call_network()
    } catch(r) {
        ...
    }
}
f()
```

Let me know if you want me to change that.